pkg/oauth/apiserver/apiserver.go

```diff
@@ -83,21 +103,25 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 		return nil, err
 	}
 
+	for hookname := range postStartHooks {
+		s.GenericAPIServer.AddPostStartHookOrDie(hookname, c.ExtraConfig.postStartHooks[hookname])
+	}
+
 	return s, nil
 }
 
-func (c *completedConfig) V1RESTStorage() (map[string]rest.Storage, error) {
+func (c *completedConfig) V1RESTStorage() (map[string]rest.Storage, map[string]genericapiserver.PostStartHookFunc, error) {
 	c.ExtraConfig.makeV1Storage.Do(func() {
-		c.ExtraConfig.v1Storage, c.ExtraConfig.v1StorageErr = c.newV1RESTStorage()
+		c.ExtraConfig.v1Storage, c.ExtraConfig.postStartHooks, c.ExtraConfig.v1StorageErr = c.newV1RESTStorage()
 	})
 
-	return c.ExtraConfig.v1Storage, c.ExtraConfig.v1StorageErr
+	return c.ExtraConfig.v1Storage, c.ExtraConfig.postStartHooks, c.ExtraConfig.v1StorageErr
```

Comment:
this func is the reason that you have the hooks in ExtraConfig?

Reply:
Mutating `c.ExtraConfig.v1Storage` (preexisting) is already strange.

Reply:
Can you clean that up? This is not right. And if we have a good reason for this unexpected state, there should be a clear comment.

Reply:
I removed those lines here.